src/views/SignTransactionLedger.vue

```diff
@@ -0,0 +1,408 @@
+            const checkoutRequest = this.request as ParsedCheckoutRequest;
+            const blockchainHeight = await network.getBlockchainHeight(); // usually instant as synced in checkout
+            // The next block is the earliest for which tx are accepted by standard miners
+            validityStartHeight = blockchainHeight + 1
+                - TX_VALIDITY_WINDOW
+                + checkoutRequest.validityDuration;
+        } else {
+            // this case get's rejected in created
+            return;
+        }
+
+        const senderAccountInfo = this.findWalletByAddress(this.senderDetails.address)!;
+        const senderAddressInfo = senderAccountInfo.accounts.get(this.senderDetails.address)!;
+
+        let signedTransaction;
+        try {
+            signedTransaction = await LedgerApi.signTransaction({
+                sender: senderAddressInfo.address,
+                recipient: this.request.recipient,
+                value: this.request.value,
+                fee: this.request.fee || 0,
+                validityStartHeight,
+                network: Config.network,
+                extraData: this.request.data,
+                flags: this.request.flags,
+            }, senderAddressInfo.path, senderAccountInfo.id);
+        } catch (e) {
+            // If cancelled, handle the exception. Otherwise just keep the error message displayed in ledger ui.
+            if (e.message.toLowerCase().indexOf('cancelled') !== -1) {
+                if (this.request.kind === RequestType.CHECKOUT) {
+                    this._back(); // user might want to choose another account or address
+                } else {
+                    this._close();
+                }
+            }
+            return;
+        }
+
+        this.shownAccountDetails = null;
+
+        let result;
+        if (this.request.kind === RequestType.CHECKOUT) {
+            this.state = SignTransactionLedger.State.SENDING_TRANSACTION;
+            result = await network.sendToNetwork(signedTransaction);
+        } else {
+            result = network.makeSignTransactionResult(signedTransaction);
+        }
+
+        this.state = SignTransactionLedger.State.FINISHED;
+        await new Promise((resolve) => setTimeout(resolve, Loader.SUCCESS_REDIRECT_DELAY));
+        this.$rpc.resolve(result);
+    }
+
+    private destroyed() {
+        const currentRequest = LedgerApi.currentRequest;
+        if (currentRequest && currentRequest.type === LedgerApi.RequestType.SIGN_TRANSACTION) {
+            currentRequest.cancel();
+        }
+    }
+
+    private get transactionValue() {
+        return Nimiq.Policy.satoshisToCoins(this.request.value);
+    }
+
+    private get transactionFee() {
+        return Nimiq.Policy.satoshisToCoins(this.request.fee || 0);
+    }
+
+    private get transactionData() {
+        return this.request.data && this.request.data.length > 0
+            ? Utf8Tools.utf8ByteArrayToString(this.request.data)
+            : null;
```

Comment:
You might want to do a check here, if the data is even valid utf8, and otherwise display it as HEX.

Reply:
See https://github.com/nimiq/accounts/issues/232